tasks/tang-key-management.yml

```diff
@@ -0,0 +1,135 @@
+---
+# This task makes sure we have nbde_server_keys_dir defined. Since
+# we are now running tasks related to key management, i.e., fetching
+# and deploying of keys, we need this directory to exist.
+# It must also be an absolute path.
+- name: Make sure nbde_server_keys_dir is set
+  fail:
+    msg: >
+      You need to specify nbde_server_keys_dir and it must be an absolute path
+  when: >
+      nbde_server_keys_dir is not defined
+      or not nbde_server_keys_dir.startswith('/')
+
+
+# This task gathers the keys from a tang server, and fills up a
+# nbde_server_keys_to_fetch variable, to be used later in the actual
+# fetching of these keys.
+- name: Gather keys to be fetched
+  find:
+    paths: "{{ __nbde_server_keydir }}"
+    hidden: yes
+    patterns:
+      - "*.jwk"
+      - ".*.jwk"
+  register: nbde_server_keys_to_fetch
+
+
+# This task fetches the keys from a tang server and places them in a
+# directory named after the host, inside nbde_server_keys_dir.
+- name: Ensure we fetch the host keys
+  when: >
+    nbde_server_fetch_keys
+    and (not nbde_server_deploy_keys)
```

Comment:
for `and` conditions, make the `when` value a list:
when:
  - nbde_server_fetch_keys | bool
  - not nbde_server_deploy_keys
```
and note that you need the `nbde_server_fetch_keys | bool` because just using `- nbde_server_fetch_keys` would make `nbde_server_fetch_keys` a "bare key"

Reply:
Good point, it even looks cleaner and more readable.